apps/src/gamelab/GameLab.js

```diff
@@ -421,11 +422,43 @@ GameLab.prototype.init = function (config) {
  * Export the project for web or use within Expo.
  * @param {Object} expoOpts
  */
-GameLab.prototype.exportApp = function (expoOpts) {
+GameLab.prototype.exportApp = async function (expoOpts) {
+  const store = getStore();
+  if (this.areAnimationsReady_()) {
+    return this.exportAppWithAnimations(store.getState().animationList, expoOpts);
+  }
+  return new Promise(resolve => {
+    // Watch store changes until all the animations are ready.
+    const unsubscribe = store.subscribe(() => {
+      if (this.areAnimationsReady_()) {
+        unsubscribe();
+        const exportResult = this.exportAppWithAnimations(
+            store.getState().animationList,
+            expoOpts
+        );
+        resolve(exportResult);
+      }
+    });
+  });
+};
```

Comment:
You've got an `async` function without an `await` here, and IMO this gets a whole lot cleaner with one.
```js
GameLab.prototype.exportApp = async function (expoOpts) {
  await this.whenAnimationsAreReady();
  return this.exportAppWithAnimations(getStore().getState().animationList, expoOpts);
};

GameLab.prototype.whenAnimationsAreReady = function () {
  return new Promise(resolve => {
    const unsubscribe = getStore().subscribe(() => {
      if (this.areAnimationsReady_()) {
        unsubscribe();
        resolve();
      }
    });
  });
}

```

Reply:
I modified your suggestion slightly to avoid subscribing to the store when `areAnimationsReady_()` returns `true` right away. This avoids requiring waiting for a change listener to fire (which may not happen if the store doesn't change).

I also modified the other place where we called `areAnimationsAreReady_()` in a similar fashion to use the new `await this.whenAnimationsAreReady();` pattern. Much cleaner in that spot as well!